lib/ansible/plugins/shell/powershell.py

```diff
@@ -77,6 +77,11 @@ def env_prefix(self, **kwargs):
         # powershell/winrm env handling is handled in the exec wrapper
         return ""
 
+    def dirname(self, path: str) -> str:
+        # Using os.path.dirname will return an empty string when path uses
+        # backslashes. Using ntpath can handle both cases for Windows.
+        return ntpath.dirname(path)
```

Comment:
should we add basename/abspath/norm to the shells or just as we find a need for them?

Reply:
I probably would do it as is, I was very reluctant to add more to the shell plugin just for this. Plus doing so here will make it harder to backport the fix as well.